File: src/ReadStorm.Domain/Models/ReadingBookmarkEntity.cs
namespace ReadStorm.Domain.Models;

/// <summary>阅读书签（按书籍 + 章节 + 页定位）。</summary>
public sealed class ReadingBookmarkEntity
{
    public string BookId { get; set; } = string.Empty;

    public int ChapterIndex { get; set; }

    public int PageIndex { get; set; }

    public string ChapterTitle { get; set; } = string.Empty;

    public string PreviewText { get; set; } = string.Empty;

    public string AnchorText { get; set; } = string.Empty;

    public string CreatedAt { get; set; } = DateTimeOffset.Now.ToString("o");

    public string Display => $"{ChapterTitle} · 第 {PageIndex + 1} 页";
}

File: pureopus/app/src/main/java/com/readstorm/app/domain/models/ReadingBookmarkEntity.kt
package com.readstorm.app.domain.models

data class ReadingBookmarkEntity(
    var bookId: String = "",
    var chapterIndex: Int = 0,
    var pageIndex: Int = 0,
    var chapterTitle: String = "",
    var previewText: String? = null,
    var anchorText: String? = null,
    var createdAt: Long = System.currentTimeMillis()
) {
    val display: String
        get() = "$chapterTitle · 第 ${pageIndex + 1} 页"
}
